lib/ansible/modules/cloud/amazon/cloudformation_facts.py

```diff
@@ -253,36 +255,37 @@ def main():
     if not HAS_BOTO3:
         module.fail_json(msg='boto3 is required.')
 
-    # Describe the stack
     service_mgr = CloudFormationServiceManager(module)
-    stack_name = module.params.get('stack_name')
-    result = {
-        'ansible_facts': { 'cloudformation': { stack_name:{} } }
-    }
-    facts = result['ansible_facts']['cloudformation'][stack_name]
-    facts['stack_description'] = service_mgr.describe_stack(stack_name)
-
-    # Create stack output and stack parameter dictionaries
-    if facts['stack_description']:
-        facts['stack_outputs'] = to_dict(facts['stack_description'].get('Outputs'), 'OutputKey', 'OutputValue')
-        facts['stack_parameters'] = to_dict(facts['stack_description'].get('Parameters'), 'ParameterKey', 'ParameterValue')
-
-    # normalize stack description API output
-    facts['stack_description'] = camel_dict_to_snake_dict(facts['stack_description'])
-    # camel2snake doesn't handle NotificationARNs properly, so let's fix that
-    facts['stack_description']['notification_arns'] = facts['stack_description'].pop('notification_ar_ns', [])
-
-    # Create optional stack outputs
-    all_facts = module.params.get('all_facts')
-    if all_facts or module.params.get('stack_resources'):
-        facts['stack_resource_list'] = service_mgr.list_stack_resources(stack_name)
-        facts['stack_resources'] = to_dict(facts.get('stack_resource_list'), 'LogicalResourceId', 'PhysicalResourceId')
-    if all_facts or module.params.get('stack_template'):
-        facts['stack_template'] = service_mgr.get_template(stack_name)
-    if all_facts or module.params.get('stack_policy'):
-        facts['stack_policy'] = service_mgr.get_stack_policy(stack_name)
-    if all_facts or module.params.get('stack_events'):
-        facts['stack_events'] = service_mgr.describe_stack_events(stack_name)
+
+    result = {'ansible_facts': {'cloudformation': {}}}
+
+    for stack_description in service_mgr.describe_stacks(module.params.get('stack_name')):
+        facts = {'stack_description': stack_description}
+        stack_name = stack_description.get('StackName')
+
+        # Create stack output and stack parameter dictionaries
+        if facts['stack_description']:
+            facts['stack_outputs'] = to_dict(facts['stack_description'].get('Outputs'), 'OutputKey', 'OutputValue')
+            facts['stack_parameters'] = to_dict(facts['stack_description'].get('Parameters'), 'ParameterKey', 'ParameterValue')
+
+        # normalize stack description API output
+        facts['stack_description'] = camel_dict_to_snake_dict(facts['stack_description'])
+        # camel2snake doesn't handle NotificationARNs properly, so let's fix that
+        facts['stack_description']['notification_arns'] = facts['stack_description'].pop('notification_ar_ns', [])
```

Comment:
Ooh, this is a nasty bug - if this wasn't in this PR,I wouldn't have spotted this. If this code happens elsewhere, it'll break when 2.4 goes out (because a now valid key is overwritten by the content of a now missing key)